decode.go

```diff
@@ -1555,3 +1580,21 @@ func makeDecodePtrFunc(t reflect.Type, opts decodeFuncOpts) decodeFunc {
 		return d.decodePointerWith(v, f)
 	}
 }
+
+// unsafeString returns a string that is only safe to use under the following conditions:
+// - b points to data on the heap
+// - the bytes pointed to by b will not be modified while the returned string exists
+// - the returned string will not be stored past the lifetime of b
+// if the method being called with an unsafe returns an error, that error may
+// contain a reference to unsafe string. if the method produces consistent results,
+// calling it again with a safe string should return an error that can be safely
+// returned to the caller.
+func unsafeString(b []byte) string {
+	if len(b) == 0 {
+		return ""
+	}
```

Comment:
If you're looking at even more performance you can use this implementation:
```go
return *(*string)(unsafe.Pointer(&b))
```
last I checked functions containing a `if` statement don't get inlined in Go.